core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -713,81 +713,118 @@ class GroupMetadataManager(brokerId: Int,
 
   // visible for testing
   private[group] def cleanupGroupMetadata(): Unit = {
-    cleanupGroupMetadata(None, groupMetadataCache.values, time.milliseconds())
+    cleanupGroupMetadata(None)
   }
 
-  def cleanupGroupMetadata(deletedTopicPartitions: Option[Seq[TopicPartition]],
```

Comment:
What I had in mind is to replace the `deletedTopicPartitions` option and `startMs` with a function which selected the topic partitions. The definition would be like this:
```scala
def cleanupGroupMetadata(groups: Iterable[GroupMetadata], selector: GroupMetdata => Seq[TopicPartition])
```

When removing expired offsets, we would do this:
```scala
val startMs = time.milliseconds()
cleanupGroupMetadata(groupMetadata.currentGroups, group => {
  group.removeExpiredOffsets(startMs)
})
```
When topics are deleted:
```scala
cleanupGroupMetadata(groupMetadata.currentGroups, group => {
  group.removeOffsets(deletedPartitions)
})
```
And finally, when the whole group is being removed:
```scala
cleanupGroupMetadata(groupMetadata.currentGroups, group => {
  group.removeAllOffsets()
})
```

Is there any reason why that wouldn't work?